src/layouts/home-assistant.ts

```diff
@@ -200,6 +200,11 @@ export class HomeAssistantAppEl extends QuickBarMixin(HassElement) {
   }
 
   protected async checkDataBaseMigration() {
+    if (!this.hass?.config?.components.includes("recorder")) {
+      this._databaseMigration = false;
+      return;
+    }
```

Comment:
This won't be the correct fix since the PR introducing it mentions: 
We can't check for recorder in hass.config.components, it won't be added until after the migration has completed

Reply:
The code mentions that people would have been able to run Home Assistant without recorder in earlier versions. Since it checked upon recorder and putted database migration on false if it doesn't exist.  That no longer happens, therefore doesn't load the frontend anymore. (But I'm not sure if people run HA without recorder...)

Reply:
From theory to practice: It definitely runs pretty decent if you have no recorder set up. Graphs and everything related is taken away. Broken on current dev.

Reply:
Line 215 should have handled this

Reply:
I've retested this, checked the code and it works with no recorder as well. All good.